src/epm.rs

```diff
@@ -31,34 +31,46 @@ use subxt::{dynamic::Value, rpc::rpc_params, tx::DynamicTxPayload};
 const EPM_PALLET_NAME: &str = "ElectionProviderMultiPhase";
 
 #[derive(Copy, Clone, Debug)]
-struct EpmConstant {
-	epm: &'static str,
+struct Constant {
+	pallet: &'static str,
 	constant: &'static str,
 }
 
-impl EpmConstant {
-	const fn new(constant: &'static str) -> Self {
-		Self { epm: EPM_PALLET_NAME, constant }
+impl Constant {
+	const fn new(pallet: &'static str, constant: &'static str) -> Self {
+		Self { pallet, constant }
 	}
 
 	const fn to_parts(&self) -> (&'static str, &'static str) {
-		(self.epm, self.constant)
+		(self.pallet, self.constant)
 	}
 
 	fn to_string(&self) -> String {
-		format!("{}::{}", self.epm, self.constant)
+		format!("{}::{}", self.pallet, self.constant)
 	}
 }
 
 /// Read the constants from the metadata and updates the static types.
 pub(crate) async fn update_metadata_constants(api: &SubxtClient) -> Result<(), Error> {
-	const SIGNED_MAX_WEIGHT: EpmConstant = EpmConstant::new("SignedMaxWeight");
-	const MAX_LENGTH: EpmConstant = EpmConstant::new("MinerMaxLength");
-	const MAX_VOTES_PER_VOTER: EpmConstant = EpmConstant::new("MinerMaxVotesPerVoter");
+	const SIGNED_MAX_WEIGHT: Constant = Constant::new(EPM_PALLET_NAME, "SignedMaxWeight");
+	const MAX_LENGTH: Constant = Constant::new(EPM_PALLET_NAME, "MinerMaxLength");
+	const MAX_VOTES_PER_VOTER: Constant = Constant::new(EPM_PALLET_NAME, "MinerMaxVotesPerVoter");
+	const EPOCH_DURATION: Constant = Constant::new("Babe", "EpochDuration");
+	const SESSIONS_PER_ERA: Constant = Constant::new("Staking", "SessionsPerEra");
+
+	// TODO: no constant is exposed directly for this, hardcoded for now.
+	const SLOT_DURATION_SECS: u32 = 6;
 
 	let max_weight = read_constant::<Weight>(api, SIGNED_MAX_WEIGHT)?;
 	let max_length: u32 = read_constant(api, MAX_LENGTH)?;
 	let max_votes_per_voter: u32 = read_constant(api, MAX_VOTES_PER_VOTER)?;
+	// session == epoch
+	let session: u32 = read_constant(api, EPOCH_DURATION)?;
+	let num_sessions_per_era: u32 = read_constant(api, SESSIONS_PER_ERA)?;
+
+	// era = session * seconds per slot * number sessions per era
+	let submission_interval_hours: u32 =
```

Comment:
This number is more or less the same all the time FYI, perhaps we can communicate it to the monitoring system in some other way other than pushing it to prometheus?

Reply:
Sure we can set this value directly in monitoring, I thought it will be good to have a universal alert for any chains (who knows how many new chains we will have). But if need to hardcode some value for monitoring in code it is better to do it on the monitoring side (it is more flexible, we don't need to wait for the release to change it)

Reply:
It's not hardcoded (except one thing that I will fix) but in practice these values "constants" as Kian said.

However, as we read it from the metadata/storage on each startup/runtime upgrade it's probably better than to handle it in monitoring.